### PR TITLE
token selection: fix loading popover position

### PR DESCRIPTION
**Pull request by taras-yemets**
Aligns the loading tooltip to the selected occurrence start.

Closes https://github.com/sourcegraph/sourcegraph/issues/45801
Closes https://github.com/sourcegraph/sourcegraph/issues/45802

| Action | Before | After | 
| -- | -- | -- | 
| show hover tooltip on 'Space' click | Screenshot 2022-12-18 at 23 01 38| Screenshot 2022-12-18 at 23 01 58|
| go to definition on 'Enter' click | Screenshot 2022-12-18 at 23 03 59 | Screenshot 2022-12-18 at 23 04 18 |
| go to definition on long mouse click | Screenshot 2022-12-18 at 23 05 34 | Remains unchanged: https://github.com/sourcegraph/sourcegraph/pull/45803#issuecomment-1357372324. | 

## Test plan
Tested manually (see screenshots).

## App preview:

- [Web](https://sg-web-taras-yemets-token-selection-fix.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | -0.00% (-0.00 kb) | -0.00% (-0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/taras-yemets/token-selection-fix-loading-tooltip-position/compare-report.html;tab=live_object) for a full comparison between the commits 5a62a34 and 8b806b4 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/blob/codemirror/token-selection/hover.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const hoveredOccurrenceField = StateField.define<Occurrence | null>({
 export async function getHoverTooltip(view: EditorView, pos: number): Promise<Tooltip | null> {
     const cmLine = view.state.doc.lineAt(pos)
     const line = cmLine.number - 1
-    const character = countColumn(cmLine.text, 1, pos - cmLine.from) - 1
+    const character = countColumn(cmLine.text, 1, pos - cmLine.from)
```
Comment (taras-yemets) [inline client/web/src/repo/blob/codemirror/token-selection/hover.ts:86]: As we don't add 1 [anymore](https://github.com/sourcegraph/sourcegraph/pull/45803/files#diff-ab02b1ddc590d01ac7067b7e6bf9fb2ab2bb153b6fe392299bafa6160d02fa03R28) when defining `position` parameter for `getHoverTooltip` function, no need to substract 1 here.
I don't expect this change to break the line and character definition logic on lines 84-86 as both `view.state.selection.main.from` and `view.state.selection.main.from+1` should be on the same line.
Tests would be definitely helpful here.

**Comment by olafurpg** [comment]
Quick review after trying this locally without reviewing the code: the position of "loading" for mouse events must appear at the mouse position because that's where the eye is focused. Example:

CleanShot 2022-12-19 at 10 37 57


For keyboard events, I'm +1 with using the occurrence start position instead of end.

**Comment by taras-yemets** [comment]
> Quick review after trying this locally without reviewing the code: the position of "loading" for mouse events must appear at the mouse position because that's where the eye is focused.

Thanks for your review, @olafurpg! Definitely makes sense. Reverted that change.

File: client/web/src/repo/blob/codemirror/token-selection/keybindings.ts [hunk 2]
```diff
@@ -43,9 +43,7 @@ const keybindings: readonly KeyBinding[] = [
             if (!occurrence) {
                 return false
             }
-            const cmLine = view.state.doc.line(occurrence.range.start.line + 1)
-            const cmPos = cmLine.from + occurrence.range.end.character
-            const spinner = new LoadingTooltip(view, cmPos)
```
Comment (taras-yemets) [inline client/web/src/repo/blob/codemirror/token-selection/keybindings.ts:48]: We get `position` (type `Position`) from `view.state.selection.main.from` on line 41 and then get `occurrence` from it. 
We don't need to calculate the tooltip position from `occurrence` as we already know the selection start position.

**Pull request merged**